Dockerize

mamaldon opened the pull request:
The first commit is just adding x86_64 support for building, but I have some questions on the second one.

Is adding rpms to the repo frowned upon? I couldn't find the answer when I looked.
I added it now since it allows us to just call docker-compose after cloning and get the containers up and running.

We can also discuss if the plugin build files belong here or if we should leave it out for users to handle.

Mariomds committed:
Add x86 build support

Mariomds committed:
Add docker files for containerization

thalerj commented on Makefile:
```diff
@@ -18,7 +18,7 @@ JAVA=/usr/bin/java
-ARCH=pp64le
+ARCH=$(shell uname -p)
 PROD=ibm-crassd
 NAME=$(PROD)-$(VER)-$(REL).$(ARCH)
```
I suggest adding the RHEL version here as well. The spec files will need to call out the different requirements for RHEL 7 requirements differently due to the change in package naming conventions.

mamaldon replied:
Added the rhel version as part of the Release after reading https://docs.fedoraproject.org/en-US/packaging-guidelines/DistTag/ and other similar suggestions. More info in the new commits

thalerj reviewed (changes requested):
I'd recommend removing the binary for openbmctool and instead recommend cloning the repo and generating it during the build process for the crassd rpm. There's a commit out for review for openbmctool to provide a makefile very similar to this one for generating an rpm or deb package binary. Here's a link to that for reference:  https://gerrit.openbmc-project.xyz/c/openbmc/openbmc-tools/+/29184

Mariomds committed:
Generalize rpmbuild files

A generic spec is now used instead of having an arch dependent one.
The generated rpm should now be noarch.
Moved the docker and compose files to an example directory, and the
Docker build will now need an rpm instead of relying on make/install.
Added a build script for the rpm generation.

Mariomds committed:
Change spec to fit general Release standards

- Rpms generated will now have the rhel version in the name
example: ibm-crassd-v-r.el8.noarch.rpm
- Sample build script will also try to build the openbmctool from github

thalerj reviewed (approved):
I agree with these changes and this looks good.
